modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/app/ApplicationConfig.scala

```diff
@@ -50,6 +54,25 @@ object ApplicationConfig {
     apiUrl  : Option[String]
   )
 
+  case class WalletsInfo(
+    address      : Address,
+    minimumAmount: Long,
```

Comment:
It should be an Amount. Also, can the minimum or maximum amount be negative? You should use a refined type there at least